digitalocean_k8s_centos_7_master.sh

```diff
@@ -50,7 +50,7 @@ PUBLICIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/public/0/ipv4/a
 echo $PRIVATEIP > /tmp/.ip
 
 # Specify node IP for kubelet.
-echo "Environment=\"KUBELET_EXTRA_ARGS=--node-ip=${PUBLICIP}\"" >> /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+echo "KUBELET_EXTRA_ARGS=--node-ip=${PUBLICIP}" > /etc/default/kubelet
```

Comment:
Looking in the [`digitalocean-cloud-controller-manager` docs](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/master/docs/getting-started.md#--cloud-providerexternal), it seems like all `kubelet` instances must include the `--cloud-provider=external` flag, so I guess that applies to master as well.

Reply:
Yeah, I did originally, but the master wouldn't start properly (it might have been a different error tho), and I'm unsure if the master-kubelet shoult have it based on this comment:

```
--cloud-provider=external
All kubelets in your cluster MUST set the flag --cloud-provider=external. kube-apiserver and kube-controller-manager must NOT set the flag --cloud-provider which will default them to use no cloud provider natively.
```
(from https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/master/docs/getting-started.md)

Reply:
The comment states that the API Server and Controller Manager must not set those flags, but as far as I get, kubelet should still set them.

I worked on this yesterday for cluster-api-provider-digitalocean, and it looks like master should set the flag indeed. I had no problems with master failing to start when flag is set.

I had some problems in the past with Weave when I wanted to update Kubicorn bootstrap scripts to include DO CCM. Master failed to become Ready because Weave would fail to start. If that's the problem, we can either debug Weave or replace it with Flannel.

While I like Weave, Flannel is much more popular on DO, as it just works, so I don't have strong opinions here. I use Flannel right now for my cluster and it works like a charm.

Here are bootstrap scripts I implemented for cluster-api-provider, that works with DO CCM. Maybe you'll find those helpful:
* [Master bootstrap script - Ubuntu 18.04 and Kubernetes 1.11.3](https://github.com/kubermatic/cluster-api-provider-digitalocean/blob/bc9ec83c1f4c31cde9fe831c579e32ad15d6ecbf/clusterctl/examples/digitalocean/provider-components.yaml.template#L138-L358)
* [Nodes bootstrap script - Ubuntu 18.04 and Kubernetes 1.11.3](https://github.com/kubermatic/cluster-api-provider-digitalocean/blob/bc9ec83c1f4c31cde9fe831c579e32ad15d6ecbf/clusterctl/examples/digitalocean/provider-components.yaml.template#L256-L359)